app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt

```diff
@@ -92,68 +76,94 @@ class ConnectionsListViewModel(
+                    )
+                )
             }
             PopupMenuItem.DELETE -> {
                 if (item.isActive) onDeleteClickEvent.postValue(ViewModelEvent(item.guid))
                 else {
-                    deleteConnectionsAndKeys(item.guid)
+                    if (item.isV2Api) {
+                        interactorV2.deleteConnectionsAndKeys(item.guid)
+                    } else {
+                        interactorV1.deleteConnectionsAndKeys(item.guid)
+                    }
                     updateViewsContent()
                 }
             }
         }
     }
 
+    override fun renameConnection(guid: String, name: String) {
+        onRenameClickEvent.postValue(
+            ViewModelEvent(
+                EditConnectionNameDialog.dataBundle(guid, name)
+            )
+        )
+    }
+
+    override fun selectSupportForConnection(guid: String) {
+        onSupportClickEvent.postValue(ViewModelEvent(guid))
+    }
+
+    override fun processDecryptedConsentsResult(result: List<ConsentData>) {
+        this.consents = result.groupBy {
+            listItemsValues.firstOrNull { viewModel ->
+                viewModel.connectionId == it.connectionId
+            }?.guid ?: ""
+        }
+        val newListItems = updateItemsWithConsentData(listItemsValues, consents)
+        listItems.postValue(newListItems)
+    }
+
     fun onEditNameResult(data: Bundle) {
         val listItem = listItemsValues.find { it.guid == data.guid } ?: return
         val newConnectionName = data.getString(KEY_NAME) ?: return
-        val itemIndex = listItemsValues.indexOf(listItem)
         if (listItem.name != newConnectionName && newConnectionName.isNotEmpty()) {
-            connectionsRepository.getByGuid(listItem.guid)?.let { connection ->
-                connectionsRepository.updateNameAndSave(connection, newConnectionName)
-                listItems.value?.get(itemIndex)?.name = newConnectionName
-                updateListItemEvent.postValue(listItem)
+            if (listItem.isV2Api) {
+                interactorV2.updateNameAndSave(listItem, newConnectionName)
+            } else {
+                interactorV1.updateNameAndSave(listItem, newConnectionName)
             }
         }
     }
 
+    override fun updateName(newConnectionName: String, listItem: ConnectionItemViewModel) {
+        val itemIndex = listItemsValues.indexOf(listItem)
+        listItems.value?.get(itemIndex)?.name = newConnectionName
+        updateListItemEvent.postValue(listItem)
+    }
+
     fun onDeleteItemResult(guid: GUID) {
         val listItem = listItemsValues.find { it.guid == guid } ?: return
-        connectionsRepository.getByGuid(listItem.guid)?.let { connection ->
-            sendRevokeRequestForConnections(listOf(connection))
-        }
-        deleteConnectionsAndKeys(listItem.guid)
+        interactorV1.sendRevokeRequestForConnections(listItem.guid)
```

Comment:
check api version before `sendRevokeRequestForConnections`.  You should not send request to both APIs